llvm/lib/LTO/LTO.cpp

```diff
@@ -1077,8 +1077,8 @@ Error LTO::addThinLTO(BitcodeModule BM, ArrayRef<InputFile::Symbol> Syms,
     for (const std::string &Name : Conf.ThinLTOModulesToCompile) {
       if (BM.getModuleIdentifier().contains(Name)) {
         ThinLTO.ModulesToCompile->insert({BM.getModuleIdentifier(), BM});
-        llvm::errs() << "[ThinLTO] Selecting " << BM.getModuleIdentifier()
-                     << " to compile\n";
+        LLVM_DEBUG(dbgs() << "[ThinLTO] Selecting " << BM.getModuleIdentifier()
```

Comment:
By default if you set `ModulesToCompile`, the thinLTO infrastructure prints out the name of every match even in release builds. This seems like it should be debug only, so making that change here. This is not required for correctness.

Reply:
We should propose this modification to the upstream.

> if you set `ModulesToCompile` in `LTOConfig` which I do

Why do we need to set `ModulesToCompile`?

Reply:
We set it so that we don't try to codegen the devicelib modules, we only want to link in their functions used by users. Without setting this, every BC module added to the thinLTO infrastructure gets codegen run on it.

Good point, I'll make a note to make an upstream PR.

Reply:
Merged upstream [here](https://github.com/llvm/llvm-project/commit/fedc7556ad5237cd11e29b8e70d412bdc39a4fa6)